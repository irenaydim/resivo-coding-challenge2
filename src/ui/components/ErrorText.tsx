import { Stack, Typography } from '@mui/material';
import ErrorRoundedIcon from '@mui/icons-material/ErrorRounded';

interface ErrorTextProps {
  children: string
}

export const ErrorText = ({ children }: ErrorTextProps) => {
  return <Stack justifyContent="center" alignItems="center" direction="row" gap={1}>
    <ErrorRoundedIcon color="error" />
    <Typography component="p" color="error.main">{children || 'Something went wrong ...'}</Typography>
  </Stack>
}
