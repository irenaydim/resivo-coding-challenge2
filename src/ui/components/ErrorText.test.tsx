import { render } from '@testing-library/react';
import { ErrorText } from './ErrorText';

jest.mock('next/router', () => require('next-router-mock'));

const errorText = 'Custom error ...'
const emptyErrorText = ''


describe('Error text', () => {
  it('should render "Custom error ..."', () => {
    const { container } = render(<ErrorText>{errorText}</ErrorText>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render "Something went wrong ..."', () => {
    const { container } = render(<ErrorText>{emptyErrorText}</ErrorText>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
