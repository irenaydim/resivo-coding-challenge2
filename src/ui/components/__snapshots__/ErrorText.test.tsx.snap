// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error text should render "Custom error ..." 1`] = `
<div
  class="css-1n7fu9l-MuiStack-root"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
    data-testid="ErrorRoundedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 11c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v4c0 .55-.45 1-1 1zm1 4h-2v-2h2v2z"
    />
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
  >
    Custom error ...
  </p>
</div>
`;

exports[`Error text should render "Something went wrong ..." 1`] = `
<div
  class="css-1n7fu9l-MuiStack-root"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
    data-testid="ErrorRoundedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 11c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v4c0 .55-.45 1-1 1zm1 4h-2v-2h2v2z"
    />
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
  >
    Something went wrong ...
  </p>
</div>
`;
